cmtc/src/main/scala/com/lunatech/cmt/client/command/Install.scala

```diff
@@ -0,0 +1,154 @@
+import com.lunatech.cmt.*
+import com.lunatech.cmt.Domain.InstallationSource.{GithubProject, LocalDirectory, ZipFile}
+import com.lunatech.cmt.Domain.{InstallationSource, StudentifiedRepo}
+import com.lunatech.cmt.Helpers.ignoreProcessStdOutStdErr
+import com.lunatech.cmt.client.Configuration
+import com.lunatech.cmt.client.cli.CmtcCommand
+import com.lunatech.cmt.core.cli.ArgParsers.installationSourceArgParser
+import com.lunatech.cmt.core.cli.enforceNoTrailingArguments
+import com.lunatech.cmt.core.validation.Validatable
+import sbt.io.IO as sbtio
+import sbt.io.syntax.*
+
+import sys.process.*
+import scala.util.{Failure, Success, Try}
+import java.io.File
+import java.net.URL
+
+object Install:
+
+  @AppName("install")
+  @CommandName("install")
+  @HelpMessage(
+    "Install a course - from either a local directory, a zip file on the local file system or a Github project")
+  final case class Options(
+      @ExtraName("s")
+      source: InstallationSource)
+
+  given Validatable[Install.Options] with
+    extension (options: Install.Options)
+      def validated(): Either[CmtError, Install.Options] =
+        options.asRight
+      end validated
+  end given
+
+  given Executable[Install.Options] with
+    extension (cmd: Install.Options)
+      def execute(configuration: Configuration): Either[CmtError, String] =
+        cmd.source match {
+          case localDirectory: LocalDirectory         => installFromLocalDirectory(localDirectory)
+          case zipFile: ZipFile                       => installFromZipFile(zipFile, configuration)
+          case githubProject @ GithubProject(_, _, _) => installFromGithubProject(githubProject, configuration)
+        }
+
+      private def installFromLocalDirectory(localDirectory: LocalDirectory): Either[CmtError, String] =
+        GenericError(
+          s"unable to install course from local directory at '${localDirectory.value.getCanonicalPath}' - installing from a local directory is not supported... yet").asLeft
+
+      private def installFromZipFile(
+          zipFile: ZipFile,
+          configuration: Configuration,
+          deleteZipAfterInstall: Boolean = false): Either[CmtError, String] =
+        sbtio.unzip(zipFile.value, configuration.coursesDirectory.value)
+        if (deleteZipAfterInstall) {
+          sbtio.delete(zipFile.value)
+        }
+        s"Unzipped '${zipFile.value.name}' to '${configuration.coursesDirectory.value.getAbsolutePath}'".asRight
+
+      private def extractTag(lsFilesTagLine: String): String =
+        lsFilesTagLine.replaceAll(""".*refs/tags/""", "")
+      private def installFromGithubProject(
+          githubProject: GithubProject,
+          configuration: Configuration): Either[CmtError, String] = {
+        val cwd = file(".").getCanonicalFile
+        // TODO This may be brittle; if the user doesn't have its git credentials set, we
+        // may want to retry using HTTP
+        // if it fails, we may still attempt to download the artefact
```

Comment:
Yeah, this is a good point but I don't think it's something we should worry about for the 2.0 release - let's just create an issue for it and link to the issue in the `TODO`

Reply:
We may take another route. Instead of trying to figure out if there's an error in the template parameter, we could limit the implementation to simply try the download. We wouldn't be able to provide detailed error info in case of an error (such a non-existing release tag), but I don't think that's a big deal as the template parameter will be supplied by the course organiser, so if it's copy/pasted, there shouldn't be an issue.

If this approach is taken, the whole retrieval of the release tags would be removed from the code.

Reply:
+1 for creating an issue for changing the approach later.